sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java

```diff
@@ -531,19 +633,41 @@ public static final FieldType row(Schema schema) {
       return FieldType.forTypeName(TypeName.ROW).setRowSchema(schema).build();
     }
 
+    /** Creates a logical type based on a primitive field type. */
+    public static final <InputT, BaseT> FieldType logicalType(
+        LogicalType<InputT, BaseT> logicalType) {
+      return FieldType.forTypeName(TypeName.LOGICAL_TYPE)
+          .setLogicalType(logicalType)
+          .build()
+          .withMetadata(LOGICAL_TYPE_IDENTIFIER, logicalType.getIdentifier())
+          .withMetadata(LOGICAL_TYPE_ARGUMENT, logicalType.getArgument());
+    }
+
     /** Returns a copy of the descriptor with metadata set. */
-    public FieldType withMetadata(@Nullable byte[] metadata) {
-      return toBuilder().setMetadata(metadata).build();
+    public FieldType withMetadata(String key, byte[] metadata) {
```

Comment:
As metadata is Map now, can we have a withMetadata(Map<String, byte[]>) as well?

Reply:
sure, done

Reply:
Is some changes not pushed? Not seeing `withMetadata(Map)` implementations, I can see comment is updated though.

Reply:
Sorry, it got messed up. Fixed now.